src/SerilogTracing/Instrumentation/ActivityInstrumentation.cs

```diff
@@ -248,6 +254,199 @@ class TextException(
+    /// <returns></returns>
+    internal static Activity? CreateReplacementActivity(
+        Activity? replace,
+        bool inheritTags,
+        bool inheritParent,
+        bool inheritFlags,
+        bool inheritBaggage
+    ) {
+        // We're only interested in the incoming parent if there is one. Switching off `inheritParent` when there isn't,
+        // prevents us from trying to override a nonexistent sampling decision a little further down. Checking
+        // `HasRemoteParent` would be useful here, but it creates problems for unit testing.
+        inheritParent = inheritParent && replace != null &&
+                        replace.ParentSpanId.ToHexString() != default(ActivitySpanId).ToHexString();
+
+        var flags = ActivityTraceFlags.None;
+        if (inheritParent && inheritFlags &&
+            replace!.ParentId != null && TryParseTraceParentHeader(replace.ParentId, out var parsed))
+        {
+            flags = parsed.Value;
+        }
+
+        var context = inheritParent && inheritFlags ?
+            new ActivityContext(
+                replace!.TraceId,
+                replace.ParentSpanId,
+                flags,
+                isRemote: true) :
+            default;
+        
+        var replacement = ReplacementActivitySource.CreateActivity(DefaultActivityName, replace?.Kind ?? ActivityKind.Internal, context);
+
+        if (replace == null)
+        {
+            return replacement;
+        }
+
+        if (replacement != null)
+        {
+            replacement.SetCustomProperty(ReplacedActivityPropertyName, replace);
+
+            if (inheritTags)
+            {
+#if FEATURE_ACTIVITY_ENUMERATETAGOBJECTS
+                foreach (var (name, value) in incoming.EnumerateTagObjects())
+#else
+                foreach (var (name, value) in replace.TagObjects)
+#endif
+                {
+                    replacement.SetTag(name, value);
+                }
+            }
+
+            if (inheritParent)
+            {
+                if (inheritFlags)
+                {
+                    // In `Trust` mode we override the local sampling decision with the remote one. We
+                    // already used the incoming trace and parent span ids through the `context` passed
+                    // to `CreateActivity`.
+                    replacement.ActivityTraceFlags = flags;
+                }
+                else
+                {
+                    replacement.SetParentId(replace.TraceId, replace.ParentSpanId, replacement.ActivityTraceFlags);
+                }
+            }
+
+            if (inheritBaggage)
+            {
+                foreach (var (k, v) in replace.Baggage)
+                {
+                    replacement.SetBaggage(k, v);
+                }
+            }
+        }
+        
+        return replacement;
+    }
+    
+    internal static bool TryParseTraceParentHeader(string traceParentHeaderValue, [NotNullWhen(true)] out ActivityTraceFlags? flags)
```

Comment:
Should move this into a dedicated type, and make the parser treat the fields more correctly.